src/Testing/FormDataParser.php

```diff
@@ -0,0 +1,162 @@
+                $formDataFile->error = UPLOAD_ERR_OK;
+                $formDataFile->tmp_name = null;
+
+                if ($formDataFile->size > $this->toBytes(ini_get('upload_max_filesize'))) {
+                    $formDataFile->error = UPLOAD_ERR_INI_SIZE;
+                } else {
+                    $tmpResource = tmpfile();
+                    if ($tmpResource === false) {
+                        $formDataFile->error = UPLOAD_ERR_CANT_WRITE;
+                    } else {
+                        $tmpResourceMetaData = stream_get_meta_data($tmpResource);
+                        $tmpFileName = $tmpResourceMetaData['uri'];
+                        if (empty($tmpFileName)) {
+                            $formDataFile->error = UPLOAD_ERR_CANT_WRITE;
+                            @fclose($tmpResource);
+                        } else {
+                            fwrite($tmpResource, $value);
+                            $formDataFile->tmp_name = $tmpFileName;
+                            $formDataFile->tmp_resource = $tmpResource;
+                        }
+                    }
+                }
+                $this->files[$headers['content-disposition']['name']] = $formDataFile;
+            } else {
+                $this->params[$headers['content-disposition']['name']] = $value;
+            }
+        }
+
+        return $this;
+    }
+
+    /**
+     * Parses body param headers
+     * @param  string  $headerContent
+     * @return array
+     */
+    protected function parseHeaders(string $headerContent): array
+    {
+        $headers = [];
+        $headerParts = preg_split('/\\R/s', $headerContent, -1, PREG_SPLIT_NO_EMPTY);
+        foreach ($headerParts as $headerPart) {
+            if (!str_contains($headerPart, ':')) {
+                continue;
+            }
+            [$headerName, $headerValue] = explode(':', $headerPart, 2);
+            $headerName = strtolower(trim($headerName));
+            $headerValue = trim($headerValue);
+            if (!str_contains($headerValue, ';')) {
+                $headers[$headerName] = $headerValue;
+            } else {
+                $headers[$headerName] = [];
+                foreach (explode(';', $headerValue) as $part) {
+                    $part = trim($part);
+                    if (!str_contains($part, '=')) {
+                        $headers[$headerName][] = $part;
+                    } else {
+                        [$name, $value] = explode('=', $part, 2);
+                        $name = strtolower(trim($name));
+                        $value = trim(trim($value), '"');
+                        $headers[$headerName][$name] = $value;
+                    }
+                }
+            }
+        }
+
+        return $headers;
+    }
+
+    /**
+     * Formatted bytes to bytes
+     * @param  string  $formattedBytes
+     * @return int
+     */
+    protected function toBytes(string $formattedBytes): int
```

Comment:
What about doing something like this? https://github.com/SergiX44/XBackBone/blob/master/app/helpers.php#L82